packages/flutter/lib/src/material/selection.dart

```diff
@@ -0,0 +1,927 @@
+  }
+
+  void _handleDragUpdate(DragUpdateDetails details) {
+    final Offset offset = details.globalPosition;
+    if (_selectionEndPosition != offset) {
+      _selectionEndPosition = offset;
+      _triggerSelectionEndEdgeUpdate();
+    }
+  }
+
+  Offset? _selectionEndPosition;
+  bool get _userDraggingSelectionEnd => _selectionEndPosition != null;
+  bool _scheduledSelectionEndEdgeUpdate = false;
+  void _triggerSelectionEndEdgeUpdate() {
+    // This method can be called when the drag is not in progress. This can
+    // happen if the the child scrollable returns SelectionResult.pending, and
+    // the selection area scheduled a selection update for the next frame, but
+    // the drag is lifted before the scheduled selection update is run.
+    if (_scheduledSelectionEndEdgeUpdate || !_userDraggingSelectionEnd)
+      return;
+    if (_selectable?.dispatchSelectionEvent(
+        SelectionEndEdgeUpdateEvent(globalPosition: _selectionEndPosition!)) == SelectionResult.pending) {
+      _scheduledSelectionEndEdgeUpdate = true;
+      SchedulerBinding.instance!.addPostFrameCallback((Duration timeStamp) {
+        _scheduledSelectionEndEdgeUpdate = false;
+        _triggerSelectionEndEdgeUpdate();
+      });
+      return;
+    }
+  }
+
+  void _stopSelectionEndEdgeUpdate([Object? object]) {
+    _scheduledSelectionEndEdgeUpdate = false;
+    _selectionEndPosition = null;
+  }
+
+  Offset? _selectionStartPosition;
+  bool get _userDraggingSelectionStart => _selectionStartPosition != null;
+  bool _scheduledSelectionStartEdgeUpdate = false;
+  void _triggerSelectionStartEdgeUpdate() {
+    // This method can be called when the drag is not in progress. This can
+    // happen if the the child scrollable returns SelectionResult.pending, and
+    // the selection area scheduled a selection update for the next frame, but
+    // the drag is lifted before the scheduled selection update is run.
+    if (_scheduledSelectionStartEdgeUpdate || !_userDraggingSelectionStart)
+      return;
+    if (_selectable?.dispatchSelectionEvent(
+        SelectionStartEdgeUpdateEvent(globalPosition: _selectionStartPosition!)) == SelectionResult.pending) {
+      _scheduledSelectionStartEdgeUpdate = true;
+      SchedulerBinding.instance!.addPostFrameCallback((Duration timeStamp) {
+        _scheduledSelectionStartEdgeUpdate = false;
+        _triggerSelectionStartEdgeUpdate();
+      });
+      return;
+    }
+  }
+
+  void _stopSelectionStartEdgeUpdate() {
+    _scheduledSelectionStartEdgeUpdate = false;
+    _selectionEndPosition = null;
+  }
+
+  void _cancelSelection() {
+    _stopSelectionEndEdgeUpdate();
+    _stopSelectionStartEdgeUpdate();
+    _selectable?.dispatchSelectionEvent(const ClearSelectionEvent());
+  }
+
+  void _selectAll([Intent? intent]) {
+    _cancelSelection();
+    _selectable?.dispatchSelectionEvent(const SelectAllSelectionEvent());
+  }
+
+  Future<void> _copy([Intent? intent]) async {
```

Comment:
How do we eventually expand support towards copy/paste for rich data types? i.e. images and things

Reply:
we can have a within app clipboard, but I am not sure how we can store them into system clipboard as they usually seems to store different data type with different API. for example windows have Clipboard.setText and Clipboard.setImage.

Reply:
I would think about how we could expand this in the future to include some metadata that would indicate the type of data to be copied. I don't necessarily think you need to worry about it initially, but it would be cool to support

Reply:
yes, I think that is a good idea. will think about how we can do that